### ISSUE TITLE
Client disconnects and freezes without any message

### ISSUE THREAD

**Issue by BlexMC** [ORelio/Minecraft-Console-Client#869]
Hello,
as already stated above, the client disconnects for no reason (it's not happening direct, it happens after some time - this time after 9h). It's like the client freezes and disconnects from the server without any message. The screen - where the client runs on - is still open but the client doesn't response to commands for example. The only way to stop that is by killing the screen as the client doesn't answers to commands anylonger.
I tried reinstalling the client now multiple times (yes, the newest build version) but the issue never got fixed..

Here's a screenshot how it looks like (don't wonder, the chat is dead at this time):
![disconnect](https://user-images.githubusercontent.com/43935665/75149123-8b303e80-5701-11ea-9516-4fb4bbd1321b.PNG)

Also, the client keeps disconnecting so often. It's like, he's online 5 minutes, reconnects, online 5 minutes, reconnects.. And it happens also sometimes, that the whole screen gets closed so I have to start the client complete new.

I already looked if someone else has this issue and it looks like this issue #753 is pretty similiar.
I hope there are any solutions or workarounds!

-BlexMC

**Comment by ORelio** [comment]
I encountered that on a Raspberry Pi once, yet another Mono issue. A simple workaround is to disconnect from time to time to perform a full restart of the client, before the Mono Framework freezes:

**wrapper.sh**
````bash
#!/bin/bash
while true; do
    mono MinecraftClient.exe
    echo "Restarting Minecraft Console Client..."
    sleep 5
done
````

**tasks.ini**
````ini
[Task]
triggerOnInterval=true
timeInterval=3600
script=exit.txt
````

**exit.txt**
````ini
log Exiting Minecraft Console Client...
exit
````

Then launch MCC using wrapper.sh. Time interval here is 3600 seconds for 1 hour, but you can also reboot MCC every night using triggerOnTime instead of triggerOnInterval.

**Comment by BlexMC** [comment]
Hello,
thanks for your answer. Generally, it looks better now but I got another issue.. Now, it happends that the client keeps getting stuck while joining the server. The message "You successfully joined the server" came but nothing else happened.

**Comment by ORelio** [comment]
Should be fixed as per #1219 thanks to @ReinforceZwei's findings 🙏
